recipes-core/systemd/systemd_%.bbappend

```diff
@@ -0,0 +1,17 @@
+PACKAGECONFIG += "networkd resolved"
+
+FILESEXTRAPATHS_prepend := "${THISDIR}/files:"
+
+SRC_URI += " file://eth.network"
+
+
+FILES_${PN} += "${sysconfdir}/systemd/network/eth.network"
+
+
+do_install_append() {
+  if ${@bb.utils.contains('PACKAGECONFIG','networkd','true','false',d)}; then
+        install -d ${D}${sysconfdir}/systemd/network
+        install -m 0755 ${WORKDIR}/eth.network ${D}${sysconfdir}/systemd/network
+  fi
+}
```

Comment:
I find it incredibly odd that there isn't already a standard way to achieve this. But no use in wondering, this is a good fix for now!